src/Microsoft.AspNetCore.SignalR.Common/Internal/MemoryBufferWriter.cs

```diff
@@ -33,8 +33,7 @@ public Memory<byte> GetMemory(int sizeHint = 0)
 
             if (Segments.Count == 0 || Position == _segmentSize)
             {
-                // TODO: Rent memory from a pool
-                Segments.Add(new Memory<byte>(new byte[_segmentSize]));
+                Segments.Add(ArrayPool<byte>.Shared.Rent(_segmentSize));
```

Comment:
What are the pros and cons of ArrayPool vs MemoryPool? https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Buffers/MemoryPool.cs

I've never used MemoryPool myself.